README.md

```diff
@@ -78,16 +78,70 @@ and this to stop it:
 curl -X DELETE http://localhost:8083/connectors/datagen-pageviews
 ```
 
+## Schema Registry
+
+The schema registry is used to encode/decode events stored in kafka.  This repository stores the schemas from the 'ec' namespace.  To modify the schemas, change the schema dsl in 'avro/dsl/*', regenerate the avsc files thru the following rake task
+```
+bundle exec rake avro:generate
+```    
+
+Handy local schema registry http interactions
+```
+GET http://localhost:8081/subjects
+GET http://localhost:8081/subjects/org.openstax.ec.nudged/versions
+DELETE http://localhost:8081/subjects/org.openstax.ec.nudged
+DELETE http://localhost:8081/subjects/org.openstax.ec.nudged?permanent=true
+```
+
 ## Swagger, Clients, and Bindings
 
 The Event Capture API is documented in the code using Swagger.  Swagger JSON can be accessed at `/api/v0/swagger`.
 
+Note: The contents of the data packet is swagger validated in addition to avro validation.  The *_swagger file for the datum lives alongside the generated avsc file in the same versioned directory. 
+
 ### Autogenerating bindings
 
 Within the baseline, we use Swagger-generated Ruby code to serve as bindings for request and response data.  Calling
 `rake openstax_swagger:generate_model_bindings[X]` will create version X request and response model bindings in `app/bindings/api/vX`.
 See the documentation at https://github.com/openstax/swagger-rails for more information.
 
+## Updating a kafka event
+For example...say you want to change the field app to "initialing_app" for the event 'nudged'
+  1. Create a new version, a new directory, for nudged datum 
+     ```
+        mkdir datum/schema/org/openstax/ec/nudged/v2
+     ```
+  1. Create a new nudged avro dsl 
+      ```
+         touch datum/dsl/org/openstax/ec/nudged/v2/nudged.rb
```

Comment:
@jpslav i had 2 tree structures because avro_builder does not work with having other files alongside the DSL (the gem author suggested us doing a PR on this sometime if we wanted this function).